JASP-Engine/JASP/R/regressionloglinearbayesian.R

```diff
@@ -16,355 +16,521 @@
+	    } else if (length(variablesInModel) > 1 || options$counts != "") {
+	        modelDefinition <- paste(dependentBase64, "~", paste(variablesInModelBase64, collapse = "+"))
+	    } else {
+	        # Nothing worked out:
+	        modelDefinition <- NULL #this model has no parameters
+	        
+	        # TODO: Tim's centralised error message 
+	        bfObject$hasErrors <- TRUE
+	        bfObject$errorMessages[[length(bfObject$errorMessages)+1]] <- list(errorMessage="variables cannot be read", 
+	                                                                           errorType="input")
+	    }
+	    
+	    # Save in object
+	    bfObject$variables <- variablesInModel
+	    
+	    # START analysis Bayesian Log Linear regression -----------------
+	    # if no error and modelDefinition is okay
+	    #
+	    if (!is.null(modelDefinition) && !isTRUE(bfObject$hasErrors)) {
+		    modelFormula <- as.formula(modelDefinition)
+
+		    if (options$counts == ""){
+	 			names(dataset)[names(dataset)== "freq"] <- dependentBase64
+		    }
+
+		    # Calculate here
+		    checkForm <<- modelFormula
+		    checkData <<- dataset
 		    
-		    logBlm.fit <- try( conting::bcct(formula=model.formula, data = dataset, prior = "SBH", n.sample=2000, a=options$priorShape, b=options$priorScale), silent = TRUE)
-		    no.burnin = 2000 * 0.2
+		    anthonyObj <- try(conting::bcct(formula=modelFormula, data = dataset, prior = "SBH", n.sample=2000, a=options$priorShape, b=options$priorScale), silent = TRUE)
+		    # anthonyObj <- conting::bcct(formula=modelFormula, data = dataset, prior = "SBH", n.sample=2000, a=options$priorShape, b=options$priorScale)
+		    bfObject$nBurnIn <- 2000 * 0.2
 		    
 		    # TODO: check the maximal number of models visited, if it's one, then sample more
-		    # dim(logBlm.fit$maximal.mod$x)[2] <- gives the max number of model visited
-		    
+		    # dim(anthonyObj$maximal.mod$x)[2] <- gives the max number of model visited, or
+		    #
+		    # Alternative, constraint the maximum number of additional iterations
+		    # In between do call.backs() <- need to figure out
+
 		    # Always do auto and then manual adds additional samples
 		    if (options$sampleMode == "manual"){
-			    logBlm.fit <- try(conting::bcctu(object = logBlm.fit, n.sample = options$fixedSamplesNumber), silent = TRUE)
-			    no.burnin = (2000 + options$fixedSamplesNumber)* 0.2
-			} 
-			
-		    if (isTryError(logBlm.fit)) {
-		        error <- .extractErrorMessage (logBlm.fit)
+		        anthonyObj <- try(conting::bcctu(object = anthonyObj, n.sample = options$fixedSamplesNumber), silent = TRUE)
+		        bfObject$nBurnIn <- (2000 + options$fixedSamplesNumber)* 0.2
+		    }
+		    
+		    # Anthony object checking
+		    if (class(anthonyObj) == "bcct") {
+		        bfObject$anthonyObj <- anthonyObj
+		    } else if (isTryError(anthonyObj)) {
+		        error <- paste0("R Package error: ", .extractErrorMessage(anthonyObj))
```

Comment:
We spoke about this yesterday a bit; I think we should either do this uniformly in every analysis or not at all. If we decide we always want to notify the user if the error is package-related (as opposed to us being dumb) I could incorporate it in the wrapping run function.